src/app/core/utils/parsers.js

```diff
@@ -117,9 +117,11 @@ const utils = {
 
     // Need to remove Z values due a incorrect addition when using
     // ol.format.WMSGetFeatureInfo readFeatures method from XML
-    // (eg. WMS getFeatureInfo); 
+    // (eg. WMS getFeatureInfo);
+    // Need to check also if feature has geometry in case of
+    // QGIS Project setting -> Add geometry to response is not checked
     if (!is3DGeometry(geometryType)){
-      features.forEach(feature => removeZValueToOLFeatureGeometry({ feature, geometryType }));
+      features.forEach(feature => feature.getGeometry() && removeZValueToOLFeatureGeometry({ feature, geometryType }));
```

Comment:
This sanity check should be done by within function you are calling, for exampe after this line:

https://github.com/g3w-suite/g3w-client/blob/b072d237d98b6f0cee9dd509663b863f14799d88/src/app/core/utils/geo.js#L22